### ISSUE TITLE
`tsImport()` not register CJS handling when `type: module` presented in root package

### ISSUE THREAD

**Issue by antfu** [privatenumber/tsx#575]
### Acknowledgements

- [X] I read the documentation and searched existing issues to avoid duplicates
- [X] I am reporting a confirmed bug (not seek debugging help)
- [X] I understand this is a collaborative project and relies on my contributions
- [X] I read and understood the [Contribution guide](https://github.com/privatenumber/tsx/blob/develop/CONTRIBUTING.md)

### Minimal reproduction URL

https://github.com/antfu/repro-tsx-tsimport-cts

### Problem & expected behavior (under 200 words)

In repro's README

### Bugs are expected to be fixed by those affected by it

- [ ] I'm interested in working on this issue

### Compensating engineering work financially will speed up resolution

- [X] I'm willing to offer financial support

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.13.1

If you appreciate this project, please consider [supporting this project by sponsoring](https://github.com/sponsors/privatenumber) :heart: :pray:

**Comment by karlhorky** [comment]
Hm, I'm seeing 0a78bfd11b20fcb6460fdcdea7d3b1daebfe3ad0 and [`tsx@4.13.1`](https://github.com/privatenumber/tsx/releases/tag/v4.13.1) break our projects:

```bash
$ tsx scripts/seedDatabase.ts
/home/runner/work/courses/courses/packages/database/util/connect.ts:2
import { cache } from 'next/dist/compiled/react/index.js';
         ^
SyntaxError: The requested module 'next/dist/compiled/react/index.js' does not provide an export named 'cache'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:134:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:217:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:3[16](https://github.com/upleveled/courses/actions/runs/9413420232/job/25930125960?pr=2494#step:12:17):24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)
Node.js v20.14.0
```

Not sure why `tsx` is having trouble resolving the `cache` export in `compiled/react/index.js`:

`compiled/react/index.js`

```js
'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react.production.js');
} else {
  module.exports = require('./cjs/react.development.js');
}
```

`

```
// ...

exports.cache = cache;
// ...
```
